projects/core/src/occ/adapters/cms/occ-cms-component.adapter.ts

```diff
@@ -32,11 +40,23 @@ export class OccCmsComponentAdapter implements CmsComponentAdapter {
     id: string,
     pageContext: PageContext
   ): Observable<T> {
-    return this.http
-      .get<T>(this.getComponentEndPoint(id, pageContext), {
-        headers: this.headers,
-      })
-      .pipe(this.converter.pipeable<any, T>(CMS_COMPONENT_NORMALIZER));
+    // TODO: (CXSPA-4886) Remove flag in the major
+    if (this.featureConfigService.isEnabled(USER_CMS_ENDPOINTS)) {
+      return this.userIdService.getUserId().pipe(
+        switchMap((userId: string) => {
+          return this.http.get<T>(
+            this.getComponentEndPoint(id, pageContext, userId),
+            {
+              headers: this.headers,
+            }
+          );
+        }),
+        this.converter.pipeable<any, T>(CMS_COMPONENT_NORMALIZER)
+      );
+    }
+    return this.http.get<T>(this.getComponentEndPoint(id, pageContext), {
```

Comment:
Originally it looks like:

```
    return this.http
      .get<T>(this.getComponentEndPoint(id, pageContext), {
        headers: this.headers,
      })
      .pipe(this.converter.pipeable<any, T>(CMS_COMPONENT_NORMALIZER));
```

the pipe part is missing. I guess we can't delete an existing code.